### ISSUE TITLE
mimxrt1050_evk:tests/drivers/kscan/kscan_api failed.

### ISSUE THREAD

**Issue by chen-png** [zephyrproject-rtos/zephyr#24700]
**To Reproduce**
1. west build -b mimxrt1050_evk -p auto tests/drivers/kscan/kscan_api
2. west flash
3. See error

**Screenshots or console output**
E: Could not enable
*** Booting Zephyr OS build zephyr-v2.2.0-1974-g4d7d3a25c127  ***
Running test suite kscan_basic_test

starting test - test_init_callback
**Cannot get KBSCAN device
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/kscan/kscan_api/src/test_kscan.c:87: test_init_callback: test_kb_callback() == TC_PASS is false**
FAIL - test_init_callback

starting test - test_control_callback
**Cannot get KBSCAN device
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/kscan/kscan_api/src/test_kscan.c:97: test_control_callback: test_disable_enable_callback() == TC_PASS is false**
FAIL - test_control_callback

Test suite kscan_basic_test failed.

PROJECT EXECUTION FAILED

**Environment (please complete the following information):**
 - OS: fedora28
 - Toolchain: zephyr-sdk-0.11.2
 - Commit ID: 4d7d3a25c1

**Comment by MaureenHelm** [comment]
I can't reproduce this on 4d7d3a2 or a more recent 22be0d8fd5902da11c9bf55a44012abc7c2cf1c1. Is this still an issue?

**Comment by chen-png** [comment]
> Is this still an issue?

yes, it still failed with latest commit ab3e778f47

*** Booting Zephyr OS build zephyr-v2.3.0-720-gab3e778f47d8  ***
Running test suite kscan_basic_test

START - test_init_callback
Cannot get KBSCAN device
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/kscan/kscan_api/src/test_kscan.c:87: test_init_callback: test_kb_callback() == TC_PASS is false
FAIL - test_init_callback

START - test_control_callback
Cannot get KBSCAN device
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/kscan/kscan_api/src/test_kscan.c:97: test_control_callback: test_disable_enable_callback() == TC_PASS is false
FAIL - test_control_callback

Test suite kscan_basic_test failed.

PROJECT EXECUTION FAILED

**Comment by MaureenHelm** [comment]
I see the problem now. There is an [LCD panel](https://www.nxp.com/design/development-boards/i-mx-evaluation-and-development-boards/4-3-lcd-panel:RK043FN02H-CT) for this board that I bet you don't have in your board farm. Without it, the kscan driver init will fail and `device_get_binding(KSCAN_DEV_NAME)` will return NULL.

@nashif can you recommend how to filter this in `testcase.yaml`? Add a harness? Not sure I want to blacklist the board since we may have the display attached in other board farms.

cc: @hakehuang

**Comment by MaureenHelm** [comment]
@nashif ping

**Comment by hakehuang** [comment]
my log shows:
```
    *** Booting Zephyr OS version 2.3.99  ***
    Running test suite kscan_basic_test
    ===================================================================
    START - test_init_callback
     PASS - test_init_callback
    ===================================================================
    START - test_control_callback
     PASS - test_control_callback
    ===================================================================
    Test suite kscan_basic_test succeeded
    ===================================================================
    PROJECT EXECUTION SUCCESSFUL

```

**Comment by MaureenHelm** [comment]
@hakehuang do you have the LCD attached?

**Comment by hakehuang** [comment]
@MaureenHelm  yes. I have the LCD attached.

**Comment by carlescufi** [comment]
Will be fixed in #28191
